lambdas/jwt-signer/tests/jwt-signer-handler.test.ts

```diff
@@ -13,203 +12,147 @@ import {
-          nbf: 1697516406,
-          persistent_session_id: "a67c497b-ac49-46a0-832c-8e7864c6d4cf",
-          redirect_uri: "https://cri.core.build.stubs.account.gov.uk/callback",
-          response_type: "code",
-          scope: "openid",
-          shared_claims: {
-            "@context": [
-              "https://www.w3.org/2018/credentials/v1",
-              "https://vocab.london.cloudapps.digital/contexts/identity-v1.jsonld",
-            ],
-            address: [
-              {
-                addressLocality: "",
-                buildingName: "",
-                buildingNumber: "",
-                postalCode: "",
-                streetName: "",
-                validFrom: "2021-01-01",
-              },
-            ],
-            birthDate: [{ value: "1948-04-23" }],
-            name: [
-              {
-                nameParts: [
-                  { type: "GivenName", value: "Jim" },
-                  { type: "FamilyName", value: "Ferguson" },
-                ],
-              },
-            ],
-          },
-          state: "diWgdrCGYnjrZK7cMPEKwJXvpGn6rvhCBteCl_I2ejg",
-          sub: "urn:fdc:gov.uk:2022:0df67954-5537-4c98-92d9-e95f0b2e6f44",
-        });
-      });
-    });
-    describe("using DIGEST signing mode", () => {
-      it("should verify a large signed jwt with claimset greater than 4096x", async () => {
-        const event: SignerPayLoad = { kid, header, claimsSet: largeClaimsSet };
-        const jwtSignerHandler = new JwtSignerHandler(
-          mockedKMSClient.prototype
-        );
-
-        const signatureBuffer = sigFormatter.joseToDer(
-          joseLargeClaimsSetSignature,
-          "ES256"
-        );
-        const spy = jest.spyOn(mockedKMSClient.prototype, "send");
-        spy.mockImplementationOnce(() =>
-          Promise.resolve({
-            Signature: signatureBuffer,
-          })
-        );
-        const signature = await jwtSignerHandler.handler(event, {} as Context);
-
-        const publicSigningVerifyingKey = await importJWK(
-          publicVerifyingJwk,
-          "ES256"
-        );
-        const { payload } = await jwtVerify(
-          `${header}.${event.claimsSet}.${signature}`,
-          publicSigningVerifyingKey,
-          {
-            algorithms: ["ES256"],
-          }
-        );
-        expect(signature).not.toBeUndefined();
-        expect(spy).toHaveBeenCalledWith(
-          expect.objectContaining({
-            input: expect.objectContaining({
-              MessageType: "DIGEST",
-            }),
-          })
-        );
-        expect(payload).toEqual(
-          JSON.parse(Buffer.from(event.claimsSet, "base64").toString())
-        );
-      });
-    });
-  });
-  describe("does not sign a jwt", () => {
-    it("should error when signature is undefined", async () => {
+const kmsClient = jest.mocked(KMSClient).prototype;
+const jwtSignerHandler = new JwtSignerHandler(kmsClient);
+
+jest.spyOn(kmsClient, "send");
```

Comment:
This is much better than before, the only thing is I think this should be in a `beforeEach`

Reply:
It doesn't need to be in `beforeEach` because the mocks get automatically reset between each test

Reply:
`const jwtSignerHandler = new JwtSignerHandler(kmsClient);` should be re-initialized for each test

I suppose you get away with it on this `https://github.com/govuk-one-login/ipv-cri-check-hmrc-api/blob/main/lambdas/jwt-signer/src/jwt-signer-handler.ts#L77`